client/client.go

```diff
@@ -138,7 +138,10 @@ func parse(agent, src, dst string) ([][]core.TransferRequest, error) {
 	// check if destination is ok
 	dstUrl, ok := remoteAgents[dst]
 	if !ok {
-		log.Println("Unable to resolve destination", dst, "known agents", remoteAgents)
+		log.WithFields(log.Fields{
+    	"Destination": dst,
+    	"known agents": remoteAgents,
```

Comment:
I prefer to keep keys CamelCase, in this case "known agents" should be replaced with Agents. Also, I think you can simply use the same semantics as log.Println() shows and it will properly formatted. Please try out and experiement with that. The main point is to get log in json format. I also want to move logger definition into config, i.e. in config we should specify which logger we want to use, e.g.

```
    // set logger settings depending on configuration
    if strings.Contains(strings.ToLower(config.LogFormatter), "json") {
        log.SetFormatter(&log.JSONFormatter{})
    }
    if strings.Contains(strings.ToLower(config.LogLevel), "info") {
        log.SetLevel(log.InfoLevel)
    }
    if strings.Contains(strings.ToLower(config.LogLevel), "warn") {
        log.SetLevel(log.WarnLevel)
    }
    if strings.Contains(strings.ToLower(config.LogLevel), "err") {
        log.SetLevel(log.ErrorLevel)
    }
    if strings.Contains(strings.ToLower(config.LogLevel), "debug") {
        log.SetLevel(log.DebugLevel)
    }
```